crates/core/component/shielded-pool/src/component/transfer.rs

```diff
@@ -335,6 +359,21 @@ async fn recv_transfer_packet_inner<S: StateWrite>(
             )
             .await
             .context("failed to mint notes in ibc transfer")?;
+
+        // update the value balance
+        let value_balance: Amount = state
+            .get(&state_key::ics20_value_balance(
+                &msg.packet.chan_on_b,
+                &denom.id(),
+            ))
+            .await?
+            .unwrap_or_else(Amount::zero);
+
+        let new_value_balance = value_balance + value.amount;
```

Comment:
I have two questions:
- should this be a `saturating_add`? Since depositing is permissionless, I could spin-up a chain with infinite issuance and make this panic
- why is this in `chan_open_init_check` vs. `recv_transfer_packet_inner`? my understanding is that we want to tally all inbound Ics20 transfers, but this method is only executed during the channel creation handshake? is that correct?

Reply:
This is in recv_transfer_packet_inner, i think it's just a rendering bug making it look like it's in open_init